packages/core/core-flows/src/definition/cart/workflows/complete-cart.ts

```diff
@@ -20,7 +21,7 @@ import { confirmVariantInventoryWorkflow } from "./confirm-variant-inventory"
 export const completeCartWorkflowId = "complete-cart"
 export const completeCartWorkflow = createWorkflow(
   completeCartWorkflowId,
-  (input: WorkflowData<any>): WorkflowData<OrderDTO> => {
+  (input: WorkflowData<any>): WorkflowResponse<WorkflowData<OrderDTO>> => {
```

Comment:
Q: should we wrap WorkflowData as part or WorkflowResponse such that you don't have to provide both always?

Reply:
Not sure. It seems like workflow return value can be a key-value pair with `WorkflowData` as the value. Referencing this code here. https://github.com/medusajs/medusa/pull/8346/files/1a54187c953e06b2b80cc7bc87ada3ea04ac0b05#diff-11ee48fe88c22283d687f2e85c805ade2ef634594e710516e603a2038fe42a29R98-R103

But if return value always have to be `WorkflowData`, then we can do as you suggested

Reply:
if you move that to a specific type, it includes all the cases
```ts
WorkflowData<TResult>
    | {
        [K in keyof TResult]:
          | WorkflowData<TResult[K]>
          | WorkflowDataProperties<TResult[K]>
          | TResult[K]
      }
```

that way when you pass the type it will allow you to return the workflow data with normal data or just normal data etc

Reply:
Done